app/billing/rest.py

```diff
@@ -141,10 +142,21 @@ def create_or_update_free_sms_fragment_limit(service_id):
     return jsonify(form), 201
 
 
-def update_free_sms_fragment_limit_data(service_id, free_sms_fragment_limit, financial_year_start=None):
+def update_free_sms_fragment_limit_data(service_id, free_sms_fragment_limit, financial_year_start):
     current_year = get_current_financial_year_start_year()
-    if financial_year_start is None or financial_year_start >= current_year:
-        dao_update_annual_billing_for_current_and_future_years(service_id, free_sms_fragment_limit)
-    else:
-        dao_create_or_update_annual_billing_for_year(service_id,
-                                                     free_sms_fragment_limit, financial_year_start)
+    if not financial_year_start:
+        financial_year_start = current_year
+
+    dao_create_or_update_annual_billing_for_year(
+        service_id,
+        free_sms_fragment_limit,
+        financial_year_start
+    )
+    # if we're trying to update historical data, don't touch other rows.
+    # Otherwise, make sure that future years will get the new updated value.
+    if financial_year_start >= current_year:
+        dao_update_annual_billing_for_current_and_future_years(
+            service_id,
+            free_sms_fragment_limit,
```

Comment:
This would work, but if the financial_year_start >= current_year, it will do the updating twice. 

We could possibly change dao_update_annual_billing_for_current_and_future_years to dao_create_and_update_annual_billing_for_current_and_future_years, so

if financial_year_start>=current_year
       # create and update multiple rows
otherwise
      # just update one row